### PR TITLE
feat(2135): Add new commands to validate and publish pipeline level template

### PR DESCRIPTION
**Pull request by aishwaryanair04**
## Context

Need new command to validate pipeline templates

## Objective

Below changes have been made:

-added new command `pipeline-template-validate`
-added function `validateTemplate` which is called by `validateJobTemplate` and `validatePipelineTemplate` in index.js
-added new operation `validatePipelineTemplate` in commands.js

## References

https://github.com/screwdriver-cd/screwdriver/issues/2135

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: commands.js [hunk 1]
```diff
@@ -201,6 +201,29 @@ const operations = {
                 });
         },
         help: 'get version from tag'
+    },
+
+    pipelineValidate: {
```
Comment (sagar1312) [inline commands.js:206]: Consider renaming
```suggestion
    validatePipelineTemplate: {
```

File: index.js [hunk 2]
```diff
@@ -25,7 +25,7 @@ function loadYaml(path) {
  */
 function validateTemplate(config) {
     const hostname = process.env.SD_API_URL || 'https://api.screwdriver.cd/v4/';
-    const url = URL.resolve(hostname, 'validator/template');
+    const url = URL.resolve(hostname, 'validator/template'); /* sd -> template-validator.js */
```
Comment (sagar1312) [inline index.js:28]:
```suggestion
    const url = URL.resolve(hostname, 'validator/template');
```

File: index.js [hunk 3]
```diff
@@ -292,6 +292,41 @@ function removeTag({ name, tag }) {
...
+
+            throw new Error(errorMessage);
+        }
+
+        return {
+            valid: true
+        };
+    });
+}
+
```
Comment (sagar1312) [inline index.js:329]: This is code is very similar to `validateTemplate()`.
Can we refactor to minimize duplicate code?

```suggestion
function validateTemplate(config, apiURL) {
  .....
  const url = URL.resolve(hostname, apiURL);
  .....
}

function validateJobTemplate(config) {
    return validateTemplate(config, 'validator/template');
}

function validatePipelineTemplate(config) {
    return validateTemplate(config, 'validator/pipelineTemplate');
}

```

File: package.json [hunk 4]
```diff
@@ -1,67 +1,68 @@
...
-    "nomnom": "^1.8.1",
-    "screwdriver-request": "^2.0.1"
-  },
-  "release": {
-    "debug": false
-  },
-  "engines": {
-    "node": ">=18.0.0"
-  }
+    "name": "screwdriver-template-main",
```
Comment (sagar1312) [inline package.json:2]: Revert the indentation to 2 spaces

File: index.js [hunk 5]
```diff
@@ -300,5 +335,6 @@ module.exports = {
     removeVersion,
     tagTemplate,
     removeTag,
-    getVersionFromTag
+    getVersionFromTag,
+    validatePipelineTemplate
```
Comment (sagar1312) [inline index.js:339]: We need to add unit tests
Refer https://github.com/screwdriver-cd/template-main/blob/master/test/index.test.js#L70

**Review by sagar1312 — CHANGES_REQUESTED** [review]

File: commands.js [hunk 6]
```diff
@@ -201,14 +201,71 @@ const operations = {
...
+                        name: publishResult.name,
+                        tag: opts.tag,
+                        version: publishResult.version
+                    })
+                )
+                .then(result => {
+                    if (!opts.json) {
+                        console.log(
+                            `Pipeline template ${result.name}@${result.version} was ` +
+                                `successfully published and tagged as ${result.tag}`
```
Comment (sagar1312) [inline commands.js:249]:
```suggestion
                            `Pipeline template ${result.name}@${result.version} was ` +
                                `successfully published`
```

File: commands.js [hunk 7]
```diff
@@ -201,14 +201,71 @@ const operations = {
                 });
         },
         help: 'get version from tag'
+    },
+
+    validatePipelineTemplate: {
```
Comment (sagar1312) [inline commands.js:206]:
```suggestion
    /* Validate pipeline template */
    validatePipelineTemplate: {
```

File: commands.js [hunk 8]
```diff
@@ -201,14 +201,71 @@ const operations = {
...
+        help: 'publish pipeline template'
     }
 };
 
 /**
  * Execute the given command by name.
  * @method run
- * @param  {String}    command name
  * @return {Object}    result of command, if any
+ * @param name
```
Comment (sagar1312) [inline commands.js:268]: Revert this doc change

File: commands.js [hunk 9]
```diff
@@ -201,14 +201,71 @@ const operations = {
...
+                    process.exit(1);
+                });
+        },
+        help: 'validate pipeline template'
+    },
+
+    publishPipelineTemplate: {
+        opts: {
+            json: { abbr: 'j', flag: true, help: 'Output result as json' },
+            tag: { abbr: 't', default: 'latest', help: 'Add template tag' }
```
Comment (sagar1312) [inline commands.js:232]:
```suggestion
            json: { abbr: 'j', flag: true, help: 'Output result as json' }
```

File: index.js [hunk 10]
```diff
@@ -292,13 +295,44 @@ function removeTag({ name, tag }) {
         };
     });
 }
+
+// eslint-disable-next-line require-jsdoc
+function publishPipelineTemplate(config) {
```
Comment (sagar1312) [inline index.js:300]: Lets refactor this similar to `validateTemplate`

**Review by sagar1312 — CHANGES_REQUESTED** [review]

File: index.js (hunk 5 above)
Comment (aishwaryanair04) [reply index.js]: Per discussion, will add tests for new commands in a follow up PR

**Review by sagar1312 — APPROVED** [review]

**Pull request merged**